bluemira/display/displayer.py

```diff
@@ -163,13 +194,113 @@ def show_cad(
             temp.modify(**kwargs)
             new_options.append(temp)
         else:
-            temp = DisplayCADOptions(**kwargs)
-            new_options.append(temp)
+            new_options.append(DisplayCADOptions(**kwargs))
+
+    part_options = [o.as_dict() for o in new_options]
+
+    polyscope_setup(
+        up_direction=kwargs.get("up_direction", "z_up"),
+        fps=kwargs.get("fps", 60),
+        aa=kwargs.get("aa", 1),
+    )
+
+    add_features(parts, part_options)
+
+    ps.show()
+
+
+def polyscope_setup(up_direction: str = "z_up", fps: int = 60, aa: int = 1):
+    """
+    Setup Polyscope default scene
+
+    Parameters
+    ----------
+    up_direction: str
+        'x_up' The positive X-axis is up.
+        'neg_x_up' The negative X-axis is up.
+        'y_up' The positive Y-axis is up.
+        'neg_y_up' The negative Y-axis is up.
+        'z_up' The positive Z-axis is up.
+        'neg_z_up' The negative Z-axis is up.
+    fps: int
+        maximum frames per second of viewer (-1 == infinite)
+    aa: int
+        anti aliasing amount, 1 is off, 2 is usually enough
+    """
+    ps.set_program_name("Bluemira Display")
+    ps.set_max_fps(fps)
+    ps.set_SSAA_factor(aa)
+    ps.set_up_dir(up_direction)
+
+    # initialize
+    ps.init()
+    ps.remove_all_structures()
 
-    shapes = [part._shape for part in parts]
-    freecad_options = [o.as_dict() for o in new_options]
 
-    cadapi.show_cad(shapes, freecad_options)
+def add_features(
+    parts: Union[BluemiraGeo, List[BluemiraGeo]],
+    options: Optional[Union[Dict, List[Dict]]] = None,
+) -> List[ps.SurfaceMesh]:
+    """
+    Grab meshes of all parts to be displayed by Polyscope
+
+    Parameters
+    ----------
+    parts: Union[BluemiraGeo, List[BluemiraGeo]]
+        parts to be displayed
+    options: Optional[Union[Dict, List[Dict]]]
+        display options
+
+    Returns
+    -------
+    meshes: List[ps.SurfaceMesh]
+        Registered Polyspline surface meshes
+
+    """
+    meshes = []
+    if not isinstance(parts, list):
+        parts = [parts]
+
+    # loop over every face adding their meshes to polyscope
+    for shape_i, (part, option) in enumerate(zip(parts, options)):
+        verts, faces = cadapi.collect_verts_faces(part._shape, option["tesselation"])
+
+        m = ps.register_surface_mesh(
+            clean_name(part.label, shape_i),
+            np.vstack(verts),
+            np.vstack(faces),
```

Comment:
`collect_verts_faces` has already stacked everything, so no need to do it again

Reply:
oops yeah, thanks!